examples/springboot-response-streaming-zip/app/src/main/java/com/amazonaws/demo/stream/controller/StreamFilesController.java

```diff
@@ -0,0 +1,73 @@
+package com.amazonaws.demo.stream.controller;
+
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+
+import java.io.*;
+import java.nio.file.Files;
+
+@RestController
+@EnableWebMvc
+public class StreamFilesController {
+
+    @RequestMapping(value = "/files/{file_name}", method = RequestMethod.GET)
+    @ResponseBody
+    public FileSystemResource getFile(@PathVariable("file_name") String fileName) {
+        return new FileSystemResource(fileName);
+    }
+
+    @GetMapping("/stream")
+    public ResponseEntity<Resource> streamFile(@RequestParam("filename") String filename) throws IOException {
+        // Load the file from the filesystem or any other source
+        File file = new File(filename);
+
+        // Check if the file exists
+        if (!file.exists()) {
```

Comment:
`filename` is from user input. We could lock it down to a specific directory to avoid security loopholes.